builtin/mainmenu/dlg_mg_flags.lua

```diff
@@ -0,0 +1,258 @@
+--Minetest
+--Copyright (C) 2015 srifqi
+--
+--This program is free software; you can redistribute it and/or modify
+--it under the terms of the GNU Lesser General Public License as published by
+--the Free Software Foundation; either version 2.1 of the License, or
+--(at your option) any later version.
+--
+--This program is distributed in the hope that it will be useful,
+--but WITHOUT ANY WARRANTY; without even the implied warranty of
+--MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--GNU Lesser General Public License for more details.
+--
+--You should have received a copy of the GNU Lesser General Public License along
+--with this program; if not, write to the Free Software Foundation, Inc.,
+--51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+local mg_first_time = true
+-- always load default
+local flag_list = {
+	-- general attributes
+	trees		= true,
+	caves		= true,
+	flat		= false,
+	dungeons	= false,
+	light		= true,
+	-- v5 mapgen
+	blobs		= true,
+	-- v6 mapgen
+	biomeblend	= true,
+	jungles		= true,
+	mudflow		= true,
+	-- v7 mapgen
+	mountains	= true,
+	ridges		= true,
+}
+
+local function update_mg_flags()
+	-- generated structure
+	-- trees, caves, flat, dungeons, light.
+	local mg_flags = ""
+	if(flag_list["trees"] == true) then
```

Comment:
This has alot of copypasta. Why not do something like this instead? 
```
local first = true
for flag, val in pairs(flag_list, flag_str) do
	flagstr = flagstr .. (first and "" or ", ") .. ((val == true) and flag)
	first = false
end
```
Put that code into a helper function, and separate the flag_list into normal flags, mgv6 flags and mgv7 flags. `mg_flags_buttonhandler` can be similarly shortened.